src/lib.rs

```diff
@@ -24,86 +45,36 @@ struct BuildState /*<'a>*/ {
-
-    for attribute in attributes {
-        let mut is_doc = false;
-        for segment in attribute.path.segments {
-            if segment.ident.to_string() == "doc" {
-                is_doc = true;
-                break;
-            }
-        }
-
-        if is_doc {
-            for token in attribute.tokens {
-                match token {
-                    syn::__private::quote::__private::TokenTree::Literal(comment) => {
-                        let comment = comment.to_string();
-                        let comment = comment[1..comment.len() - 1].trim();
-                        comments.push(comment.to_string());
-                    }
-                    _ => { /* Do nothing */ }
+impl BuildState {
+    fn write_comments(&mut self, comments: &Vec<String>, indentation_amount: i8) {
+        let indentation = utils::build_indentation(indentation_amount);
+        match comments.len() {
+            0 => (),
+            1 => self
+                .types
+                .push_str(&format!("{}/** {} */\n", indentation, &comments[0])),
+            _ => {
+                self.types.push_str(&format!("{}/**\n", indentation));
+                for comment in comments {
+                    self.types
+                        .push_str(&format!("{} * {}\n", indentation, &comment))
                 }
+                self.types.push_str(&format!("{} */\n", indentation))
             }
         }
     }
-
-    comments
-}
-
-fn build_indentation(indentation_amount: i8) -> String {
-    let mut indent = "".to_string();
-    for _ in 0..indentation_amount {
-        indent.push(' ');
-    }
-    indent
-}
-
-fn write_comments(state: &mut BuildState, comments: &Vec<String>, indentation_amount: i8) {
-    let indentation = build_indentation(indentation_amount);
-    match comments.len() {
-        0 => (),
-        1 => state.types.push_str(&format!("{}/** {} */\n", indentation, &comments[0])),
-        _ => {
-            state.types.push_str(&format!("{}/**\n", indentation));
-            for comment in comments {
-                state.types.push_str(&format!("{} * {}\n", indentation, &comment))
-            }
-            state.types.push_str(&format!("{} */\n", indentation))
-        }
-    }
 }
 
-fn extract_struct_generics(s: ItemStruct) -> String {
-    let mut generic_params: Vec<String> = vec![];
-
-    for generic_param in s.generics.params {
-        match generic_param {
-            GenericParam::Type(ty) => {
-                generic_params.push(ty.ident.to_string())
-            },
-            _ => {}
-        }
-    }
-
-    if generic_params.len() == 0 {
-        "".to_string()
-    } else {
-        format!("<{list}>", list = generic_params.join(", "))
-    }
-}
-
-fn process_rust_file(debug: bool, input_path: PathBuf, state: &mut BuildState) {
+fn process_rust_file(
+    debug: bool,
+    input_path: PathBuf,
+    state: &mut BuildState,
+    uses_typeinterface: bool,
+) {
+    dbg!(uses_typeinterface);
```

Comment:
Looks like you were debugging this :)

We should keep this behind a debug flag or remove it

```suggestion
    if debug { dbg!(uses_typeinterface); }
```

Reply:
Sorry missed that before committing was meant to remove.